test/test_prototype_transforms_kernels.py

```diff
@@ -0,0 +1,435 @@
+            image_size=bounding_box_loader.image_size,
+            angle=angle,
+            translate=(translate, translate),
+            scale=scale,
+            shear=(shear, shear),
+            center=center,
+        )
+
+
+KERNEL_INFOS.extend(
+    [
+        KernelInfo(
+            F.affine_image_tensor,
+            sample_inputs_fn=sample_inputs_affine_image_tensor,
+            reference_fn=pil_reference_wrapper(F.affine_image_pil),
+            reference_inputs_fn=reference_inputs_affine_image_tensor,
+            atol=1e-5,
+            rtol=0,
+            agg_method="mean",
+        ),
+        KernelInfo(
+            F.affine_bounding_box,
+            sample_inputs_fn=sample_inputs_affine_bounding_box,
+            reference_fn=reference_affine_bounding_box,
+            reference_inputs_fn=reference_inputs_affine_bounding_box,
+        ),
+    ]
+)
+
+
+class TestCommon:
+    sample_inputs = pytest.mark.parametrize(
+        ("info", "args_kwargs"),
+        [
+            pytest.param(info, args_kwargs, id=f"{info.kernel.__name__}")
+            for info in KERNEL_INFOS
+            for args_kwargs in info.sample_inputs_fn()
+        ],
+    )
+
+    @sample_inputs
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    def test_scripted_vs_eager(self, info, args_kwargs, device):
+        kernel_eager = info.kernel
+        try:
+            kernel_scripted = torch.jit.script(kernel_eager)
+        except Exception as error:
+            raise AssertionError("Trying to `torch.jit.script` the kernel raised the error above.") from error
+
+        args, kwargs = args_kwargs.load(device)
+
+        actual = kernel_scripted(*args, **kwargs)
+        expected = kernel_eager(*args, **kwargs)
+
+        assert_close(actual, expected, **info.closeness_kwargs)
+
+    @sample_inputs
+    @pytest.mark.parametrize("device", cpu_and_gpu())
+    def test_batched_vs_single(self, info, args_kwargs, device):
+        def unbind_batch_dims(batched_tensor, *, data_dims):
+            if batched_tensor.ndim == data_dims:
+                return batched_tensor
+
+            return [unbind_batch_dims(t, data_dims=data_dims) for t in batched_tensor.unbind(0)]
+
+        def stack_batch_dims(unbound_tensor):
+            if isinstance(unbound_tensor[0], torch.Tensor):
+                return torch.stack(unbound_tensor)
+
+            return torch.stack([stack_batch_dims(t) for t in unbound_tensor])
+
+        (batched_input, *other_args), kwargs = args_kwargs.load(device)
+
+        feature_type = features.Image if features.is_simple_tensor(batched_input) else type(batched_input)
+        # This dictionary contains the number of rightmost dimensions that contain the actual data.
+        # Everything to the left is considered a batch dimension.
+        data_dims = {
+            features.Image: 3,
+            features.BoundingBox: 1,
+            features.SegmentationMask: 3,
+        }.get(feature_type)
```

Comment:
To be able to test the transform with batched vs. single inputs, we need to know where the batch dimensions start. For masks this cannot be done, since detection masks use the last three dims for a single mask while segmentation masks use the last two.